### PR TITLE
Revert the use of libsodium for KES.

### PR DESCRIPTION
**Pull request by nc6**
This was causing a segfault in downstream code. Reverting while we do
further investigation.

### CODE REVIEW

File: ouroboros-consensus-shelley-test/src/Test/ThreadNet/Infra/Shelley.hs [hunk 1]
```diff
@@ -170,8 +170,7 @@ genCoreNode startKESPeriod = do
     delKey <- genKeyDSIGN <$> genSeed (seedSizeDSIGN (Proxy @(DSIGN c)))
     stkKey <- genKeyDSIGN <$> genSeed (seedSizeDSIGN (Proxy @(DSIGN c)))
     vrfKey <- genKeyVRF   <$> genSeed (seedSizeVRF   (Proxy @(VRF   c)))
-    kesKey <- genKeyKES . mlsbFromByteString
-                          <$> genBytes (seedSizeKES  (Proxy @(KES   c)))
+    kesKey <- genKeyKES   <$> genSeed (seedSizeKES  (Proxy @(KES   c)))
```
Comment (mrBliss) [inline ouroboros-consensus-shelley-test/src/Test/ThreadNet/Infra/Shelley.hs:173]:
```suggestion
    kesKey <- genKeyKES   <$> genSeed (seedSizeKES   (Proxy @(KES   c)))
```
Am I not predictable? :slightly_smiling_face:

**Review by mrBliss — APPROVED** [review]
bors merge

**Comment by nc6** [comment]
bors retry

**Comment by mrBliss** [comment]
The golden tests for the Allegra and Mary blocks fail. What changed?

(To update the golden tests: delete the `ouroboros-consensus-cardano-test/test/golden` folder and run them again)

**Comment by nc6** [comment]
> The golden tests for the Allegra and Mary blocks fail. What changed?
> 
> (To update the golden tests: delete the `ouroboros-consensus-cardano-test/test/golden` folder and run them again)

I guess various things will have changed, the serialisation formats haven't stabilised yet for Mary and Allegra.

**Review by mrBliss — APPROVED** [review]

**Comment by mrBliss** [comment]
bors merge

**Pull request merged**